handlers/progress_report_resource.go

```diff
@@ -0,0 +1,57 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/companieshouse/insolvency-api/service"
+	"github.com/gorilla/mux"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/insolvency-api/dao"
+	"github.com/companieshouse/insolvency-api/models"
+	"github.com/companieshouse/insolvency-api/transformers"
+	"github.com/companieshouse/insolvency-api/utils"
+)
+
+// HandleCreateProgressReport receives a progress report to be stored against the Insolvency case
+func HandleCreateProgressReport(svc dao.Service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		transactionID, validTransactionId := utils.HandleTransactionIdExistsValidation(w, req, utils.GetTransactionIDFromVars(mux.Vars(req)))
+		if !validTransactionId {
+			return
+		}
+
+		log.InfoR(req, fmt.Sprintf("start POST request for submit progress report with transaction id: %s", transactionID))
+
+		isTransactionClosed, err, httpStatus := service.CheckIfTransactionClosed(transactionID, req)
+		err, validTransactionNotClosed := utils.HandleTransactionNotClosedValidation(w, req, transactionID, isTransactionClosed, err, httpStatus)
+		if !validTransactionNotClosed {
+			return
+		}
+
+		//todo replace using generics when GO version 1.18+
+		var request models.ProgressReport
+		err = json.NewDecoder(req.Body).Decode(&request)
+
+		// Request body failed to get decoded
+		if !utils.HandleBodyDecodedValidation(w, req, transactionID, err) {
+			return
+		}
+
+		//todo replace using generics when GO version 1.18+
+		progressReportDao := transformers.ProgressReportResourceRequestToDB(&request)
```

Comment:
Check if not nil should be considered I think.

Reply:
not sure how to force failure in a test due to strong typing?